Meta/CMake/flac_spec_tests.cmake

```diff
@@ -0,0 +1,24 @@
+if(INCLUDE_FLAC_SPEC_TESTS)
+    if (CMAKE_PROJECT_NAME STREQUAL "SerenityOS")
+        set(SOURCE_DIR "${SerenityOS_SOURCE_DIR}")
+    else()
+        set(SOURCE_DIR "${SERENITY_PROJECT_ROOT}")
+    endif()
+    set(FLAC_SPEC_TEST_GZ_URL https://github.com/ietf-wg-cellar/flac-test-files/archive/refs/heads/main.tar.gz)
+
+    set(FLAC_TEST_PATH ${CMAKE_BINARY_DIR}/Tests/LibAudio/FLAC CACHE PATH "Location of FLAC tests")
+    set(FLAC_SPEC_TEST_GZ_PATH ${FLAC_TEST_PATH}/flac-spec-testsuite.tar.gz)
+    set(FLAC_SPEC_TEST_PATH ${FLAC_TEST_PATH}/SpecTests)
+
+    if(NOT EXISTS ${FLAC_SPEC_TEST_GZ_PATH})
+        message(STATUS "Downloading the IETF CELLAR FLAC testsuite from ${FLAC_SPEC_TEST_GZ_URL}...")
+        file(DOWNLOAD ${FLAC_SPEC_TEST_GZ_URL} ${FLAC_SPEC_TEST_GZ_PATH} INACTIVITY_TIMEOUT 10)
+    endif()
+
+    if(EXISTS ${FLAC_SPEC_TEST_GZ_PATH} AND NOT EXISTS ${FLAC_SPEC_TEST_PATH})
+        file(MAKE_DIRECTORY ${FLAC_SPEC_TEST_PATH})
+        message(STATUS "Extracting the FLAC testsuite from ${FLAC_SPEC_TEST_GZ_PATH}...")
+        execute_process(COMMAND "${TAR_TOOL}" -xzf ${FLAC_SPEC_TEST_GZ_PATH} -C ${FLAC_TEST_PATH})
+        file(RENAME "${FLAC_TEST_PATH}/flac-test-files-main/subset" ${FLAC_SPEC_TEST_PATH})
+    endif()
```

Comment:
Similarly, we should check for errors here. We don't have a common utility for this one yet, but check the `extract_tzdb_file` function in `Meta/CMake/time_zone_data.cmake`.

(It's annoying, but CMake won't abort the build by default if things like `execute_process` fail, we have to be explicit)